src/plugins/uploader.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+const SocketIOFileUploadServer = require("socketio-file-upload/server");
+const Helper = require("../helper");
+const path = require("path");
+const fsextra = require("fs-extra");
+const fs = require("fs");
+const fileType = require("file-type");
+const readChunk = require("read-chunk");
+const crypto = require("crypto");
+const isUtf8 = require("is-utf8");
+
+const whitelist = [
+	"text/plain",
+
+	"image/png",
+	"image/jpeg",
+	"image/jpg",
+	"image/gif",
+	"image/bmp",
+	"image/webp",
+
+	"audio/midi",
+	"audio/x-midi",
+	"video/mp4",
+	"audio/mpeg",
+	"audio/mpeg3",
+	"audio/x-mpeg",
+	"audio/x-mpeg-3",
+	"audio/wave",
+	"audio/wav",
+	"audio/x-wav",
+	"audio/x-pn-wav",
+	"audio/webm",
+	"video/webm",
+	"audio/ogg",
+	"video/ogg ",
+	"application/ogg",
+];
+
+class Uploader {
+	constructor(client, socket) {
+		const siofu = new SocketIOFileUploadServer();
+
+		siofu.uploadValidator = function(event, callback) {
+			const folder = path.join(Helper.getFileUploadPath(), ".tmp");
+			fsextra.ensureDir(folder, (err) => {
+				if (err) {
+					callback(err);
+				} else {
+					siofu.dir = folder;
+					callback(true);
+				}
+			});
+		};
+
+		siofu.on("complete", (data) => {
+			handleSaving(data).then((randomName) => {
+				const randomFileName = randomName;
+				const slug = data.file.name;
```

Comment:
I think `file.base` should be the cleaned up and safer name, so we probably should use that.

Reply:
We don't want file extensions?